lib/Controller/DisplayGroup.php

```diff
@@ -1736,4 +1749,181 @@ public function command($displayGroupId)
+     */
+    public function copyForm($displayGroupId)
+    {
+        // Create a form out of the config object.
+        $displayGroup = $this->displayGroupFactory->getById($displayGroupId);
+
+        if ($this->getUser()->userTypeId != 1 && $this->getUser()->userId != $displayGroup->userId)
+            throw new AccessDeniedException(__('You do not have permission to delete this profile'));
+
+        $this->getState()->template = 'displaygroup-form-copy';
+        $this->getState()->setData([
+            'displayGroup' => $displayGroup
+        ]);
+    }
+
+    /**
+     * Copy Display Group
+     * @param int $displayGroupId
+     * @throws \Xibo\Exception\XiboException
+     *
+     * @SWG\Post(
+     *  path="/displaygroup/{displayGroupId}/copy",
+     *  operationId="displayGroupCopy",
+     *  tags={"displaygroup"},
+     *  summary="Copy Display Group",
+     *  description="Copy an existing Display Group",
+     *  @SWG\Parameter(
+     *      name="displayGroupId",
+     *      in="path",
+     *      description="The Display Group ID",
+     *      type="integer",
+     *      required=true
+     *   ),
+     *  @SWG\Parameter(
+     *      name="displayGroup",
+     *      in="formData",
+     *      description="The name for the copy",
+     *      type="string",
+     *      required=true
+     *   ),
+     *  @SWG\Parameter(
+     *      name="description",
+     *      in="formData",
+     *      description="The description for the copy",
+     *      type="string",
+     *      required=false
+     *   ),
+     *  @SWG\Parameter(
+     *      name="copyMembers",
+     *      in="formData",
+     *      description="Flag indicating whether to copy all display and display group members",
+     *      type="integer",
+     *      required=false
+     *   ),
+     *  @SWG\Parameter(
+     *      name="copyAssignments",
+     *      in="formData",
+     *      description="Flag indicating whether to copy all layout and media assignments",
+     *      type="integer",
+     *      required=false
+     *   ),
+     *  @SWG\Parameter(
+     *      name="copyTags",
+     *      in="formData",
+     *      description="Flag indicating whether to copy all tags",
+     *      type="integer",
+     *      required=false
+     *   ),
+     *  @SWG\Response(
+     *      response=201,
+     *      description="successful operation",
+     *      @SWG\Schema(ref="#/definitions/DisplayGroup"),
+     *      @SWG\Header(
+     *          header="Location",
+     *          description="Location of the new record",
+     *          type="string"
+     *      )
+     *  )
+     * )
+     */
+    public function copy($displayGroupId)
+    {
+        // get display group object
+        /** @var \Xibo\Entity\DisplayGroup $displayGroup */
+        $displayGroup = $this->displayGroupFactory->getById($displayGroupId);
+
+        // get an array of assigned displays
+        /** @var Display[] $membersDisplays */
```

Comment:
These type hints shouldn't be necessary if the Factory has the correct `@returns` docblock on it